### PR TITLE
add stop token bits, receiver queries, a manual event loop, and a single-thread context

### PR DESCRIPTION
**Pull request by ericniebler**

### CODE REVIEW

File: examples/schedulers/manual_event_loop.hpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+#include <mutex>
+#include <type_traits>
+
+namespace example {
+  struct manual_event_loop;
+
+  namespace detail {
+    struct task_base {
+      void execute() noexcept {
+        this->execute_(this);
```
Comment (kirkshoop) [inline examples/schedulers/manual_event_loop.hpp:32]: `execute_` does not match the spelling `_execute` on line 41

File: examples/schedulers/manual_event_loop.hpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      if (head_ == nullptr) {
+        tail_ = nullptr;
+      }
+      lock.unlock();
+      task->execute();
+      lock.lock();
+    }
+  }
+
+  inline void manual_event_loop::stop() {
```
Comment (kirkshoop) [inline examples/schedulers/manual_event_loop.hpp:146]: probably no change from libunifex, but stop doesn't appear to do the right thing. I don't see the code in run that loops through all the tasks invoking set_done()

**Review by kirkshoop — APPROVED** [review]

File: examples/schedulers/manual_event_loop.hpp (hunk 1 above)
Comment (ericniebler) [reply examples/schedulers/manual_event_loop.hpp:32]: It's not referring to that function. It's referring to the `task_base::execute_` data member on line 36.

File: examples/schedulers/manual_event_loop.hpp (hunk 2 above)
Comment (ericniebler) [reply examples/schedulers/manual_event_loop.hpp:146]: It doesn't. It only stops when the queue has drained.

Comment (kirkshoop) [reply examples/schedulers/manual_event_loop.hpp:146]: It is a bit confusing because stop means cancel elsewhere and it means exit run when empty here.

Comment (ericniebler) [reply examples/schedulers/manual_event_loop.hpp:146]: Perhaps I should call it `request_stop()`?

**Pull request merged**